pandas/core/array_algos/masked_reductions.py

```diff
@@ -0,0 +1,47 @@
+"""
+masked_reductions.py is for reduction algorithms using a mask-based approach
+for missing values.
+"""
+
+import numpy as np
+
+from pandas._libs import missing as libmissing
+from pandas.compat.numpy import _np_version_under1p17
+
+from pandas.core.nanops import _below_min_count
+
+
+def sum(
```

Comment:
can we avoid overriding a builtin-in name?

Reply:
> can we avoid overriding a builtin-in name?

Do we care? In practice this is only used as `masked_reductions.sum(..)`, so there is no risk of confusing it (eg numpy also has a sum ..).

Reply:
I also don't have a problem with renaming it to eg `masked_sum`, though (but all together it then becomes a bit long `masked_reductions.masked_sum(..)`)

Reply:
`nansum`?  Not a strong enough opinion to be a blocker.

Reply:
There are no nan's involved here ;), that's exactly the difference with nanops

Reply:
ha, sounds good